pkg/config/validation/validation.go

```diff
@@ -716,6 +724,13 @@ func validateExportTo(namespace string, exportTo []string, isServiceEntry bool)
 			}
 		}
 
+		// Make sure workloadSelector based destination rule does not use exportTo other than current namespace
+		if _, exists := exportToMap[namespace]; !exists {
+			if isDestinationRuleWithWs {
+				errs = appendErrors(errs, fmt.Errorf("destination rule with workload selector cannot have exportTo beyond current namespace"))
+			}
+		}
+
```

Comment:
This looks a little hard to read, actually isDestinationRuleWithWs implicitly means dr with selector